recipes/nodejs/all/conanfile.py

```diff
@@ -1,48 +1,82 @@
         return os.path.join(self.source_folder, "source_subfolder")
 
     @property
     def _nodejs_arch(self):
-        if str(self.settings.os) == "Linux":
-            if str(self.settings.arch).startswith("armv7"):
+        if str(self.info.settings.os) == "Linux":
+            if str(self.info.settings.arch).startswith("armv7"):
                 return "armv7"
-            if str(self.settings.arch).startswith("armv8") and "32" not in str(self.settings.arch):
+            if str(self.info.settings.arch).startswith("armv8") and "32" not in str(self.info.settings.arch):
                 return "armv8"
-        return str(self.settings.arch)
+        return str(self.info.settings.arch)
+
+    @property
+    def _glibc_version(self):
+        buff = StringIO()
+        self.run(['ldd', '--version'], output=buff)
+        buff = re.search('GLIBC \d{1,3}.\d{1,3}', buff.getvalue())
+        return str(buff.group()[6:])
 
     def validate(self):
-        if (not (self.version in self.conan_data["sources"]) or
-            not (str(self.settings.os) in self.conan_data["sources"][self.version]) or
-            not (self._nodejs_arch in self.conan_data["sources"][self.version][str(self.settings.os)] ) ):
+        if not self.version in self.conan_data["sources"] or \
+           not str(self.info.settings.os) in self.conan_data["sources"][self.version] or \
+           not self._nodejs_arch in self.conan_data["sources"][self.version][str(self.info.settings.os)]:
             raise ConanInvalidConfiguration("Binaries for this combination of architecture/version/os not available")
 
+        if Version(self.version) >= "18.0.0":
+            if str(self.info.settings.compiler) == "gcc" and Version(self.info.settings.compiler.version) < "8.3":
+                raise ConanInvalidConfiguration("Binaries for this combination of architecture/version/os not available")
+
+            if str(self.info.settings.os) == "Linux":
+                if Version(self._glibc_version) < '2.27':
+                    raise ConanInvalidConfiguration("Binaries for this combination of architecture/version/os not available")
+
     def build(self):
-        tools.get(**self.conan_data["sources"][self.version][str(self.settings.os)][self._nodejs_arch], destination=self._source_subfolder, strip_root=True)
+        pass
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version][str(self.info.settings.os)][self._nodejs_arch], destination=self._source_subfolder, strip_root=True)
 
     def package(self):
-        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
-        self.copy(pattern="*", src=os.path.join(self._source_subfolder, "bin"), dst="bin")
-        self.copy(pattern="node.exe", src=self._source_subfolder, dst="bin")
-        self.copy(pattern="npm", src=self._source_subfolder, dst="bin")
-        self.copy(pattern="npx", src=self._source_subfolder, dst="bin")
+        if Version(__version__) < '2.0.0':
```

Comment:
This is not required you can just use the new helper for 1.x

Reply:
Shall be sloved by 2121d7acb9b0ef42d2d9305a112990479136328f